### PR TITLE
Date.Time.t -> Date.time

### PR DESCRIPTION
**Pull request by cknitt**
The module Time seems overkill.

### CODE REVIEW

File: src/Core__Date.res [hunk 1]
```diff
@@ -1,14 +1,12 @@
 type t = Js.Date.t
 
-module Time = {
-  type t = float
-}
+type time = float
```
Comment (glennsl) [inline src/Core__Date.res:3]: This is a confusing name for what this actually is. It isn't any more or less a "time" than `t`, just a different representation. The closest proper name for the format I think is "Unix time".

Comment (cknitt) [reply src/Core__Date.res:3]: To be explicit, maybe it could be called `millisSinceEpoch`.

Comment (glennsl) [reply src/Core__Date.res:3]: Yeah, I think that would be better. Since it's no longer a module I don't think it matters much that it's a bit wordy, although ideally I think it also ought to be a private type so that it's distinct from a `float`, which suggests it should be a module so that it can also have a conventional constructor function. But communicating the concept clearly is more important.

Comment (zth) [reply src/Core__Date.res:3]: Can also add more information about what it is exactly in a docstring.
